html/js/guiutils/Tools.js

```diff
@@ -54,13 +54,36 @@ GTE.UI = (function (parentModule) {
             case GTE.MODES.PLAYER_ASSIGNMENT:
                 buttonToSwitch = "button-player-" + this.activePlayer;
                 break;
+            case GTE.MODES.MERGE:
+                if (this.ableToSwitchToISetMode()) {
+                    buttonToSwitch = "button-merge";
+                    // If iset tools have never been chosen
+                    if (!this.isetToolsRan) {
+                        // Assign singleton isets to each node with no iset
+                        GTE.tree.initializeISets();
```

Comment:
initializeISets gets all the nodes across the tree and creates a singleton information set for each of them. From that point on, each new node will belong to a singleton information set that it's automatically created. There is no need to call initializeISets again because all nodes in the tree will belong to an information set.

In summary, initializeISets is ran only once and it creates singleton isets (including its moves) for each node. From that point on, every node will be created along with a singleton information set